### PR TITLE
run Rector on some src dirs, add types in function signatures and returns

### PR DESCRIPTION
**Pull request by beejeebus**
This is part 1 of n from #4675.

It might be good enough to merge now, and we can discuss next bits.

### CODE REVIEW

File: src/Boot/BootstrapManager.php [hunk 1]
```diff
@@ -410,15 +390,12 @@ protected function maxPhaseLimit($bootstrap_str)
     /**
      * Bootstrap to the specified phase.
      *
-     * @param int $max_phase_index
```
Comment (weitzman) [inline src/Boot/BootstrapManager.php:413]: Shouldn’t we only remove the type hint and not the whole line? The description lacks context without the `@param` line above.

File: src/Config/ConfigLocator.php [hunk 2]
```diff
@@ -160,15 +158,15 @@ public function sources()
      *
      * @return string[]
      */
-    public function configFilePaths()
+    public function configFilePaths(): array
```
Comment (weitzman) [inline src/Config/ConfigLocator.php:161]: Should we change these array hints to Iterable per https://steemit.com/php/@crell/php-never-type-hint-on-arrays?

Comment (beejeebus) [reply src/Config/ConfigLocator.php:161]: Can we do that as a follow up? My intent is to keep the non-scripted changes low, and to do multiple passes with multiple PRs.

File: src/Boot/BootstrapManager.php (hunk 1 above)
Comment (beejeebus) [reply src/Boot/BootstrapManager.php:413]: I'll see if there's a way to teach rector to leave those alone.

File: src/Config/ConfigLocator.php (hunk 2 above)
Comment (weitzman) [reply src/Config/ConfigLocator.php:161]: Yes, makes sense.

File: src/Boot/BootstrapManager.php (hunk 1 above)
Comment (beejeebus) [reply src/Boot/BootstrapManager.php]: I found a way to make this stop, with a couple of one-line hacks to Rector.

File: src/Boot/DrupalBoot8.php [hunk 3]
```diff
@@ -36,23 +36,23 @@ class DrupalBoot8 extends DrupalBoot implements AutoloaderAwareInterface
     /**
      * @return \Symfony\Component\HttpFoundation\Request
      */
-    public function getRequest()
+    public function getRequest(): \Symfony\Component\HttpFoundation\Request
```
Comment (weitzman) [inline src/Boot/DrupalBoot8.php:39]: When we add a return hint like this, we should remove the Doxygen when the doxygen has no description (e.g. this method). Is that possible?

Comment (beejeebus) [reply src/Boot/DrupalBoot8.php:39]: Let me poke at it. The "fix" for your initial request was to comment out the calls to remove docblock tags inside Rector, which is a very brute-force approach.

To be smahhter, I think we'll need to override these steps with our own, or contribute some patches upstream.

I'm ok to work on the longer steps if you think it's a worthwhile investment for Drush.

Comment (weitzman) [reply src/Boot/DrupalBoot8.php:39]: Yeah, I think its worthwhile ... Would be great if you could put in a few hours and see how far you get. Thanks!

Comment (weitzman) [reply src/Boot/DrupalBoot8.php:39]: If you arent available for more work on this, I'm OK with merging what we have.

**Comment by weitzman** [comment]
We did it in a couple PRs. #4959  and #4963

**Pull request closed without merging**